### PR TITLE
chore(all): use destructuring in conformance

### PR DESCRIPTION
**Pull request by mayeul-zama**

### CODE REVIEW

File: tfhe/src/core_crypto/entities/lwe_ciphertext.rs [hunk 1]
```diff
@@ -761,9 +761,14 @@ where
     type ParameterSet = LweCiphertextParameters<C::Element>;
 
     fn is_conformant(&self, lwe_ct_parameters: &LweCiphertextParameters<C::Element>) -> bool {
+        let Self {
+            data: _,
```
[the remark this answers is not in the document]
Comment (mayeul-zama) [reply tfhe/src/core_crypto/entities/lwe_ciphertext.rs]: Done

File: tfhe/src/shortint/ciphertext/compressed_ciphertext_list.rs [hunk 2]
```diff
@@ -23,37 +23,43 @@ impl ParameterSetConformant for CompressedCiphertextList {
...
     fn is_conformant(&self, params: &CompressedCiphertextConformanceParams) -> bool {
-        let len = self.modulus_switched_glwe_ciphertext_list.len();
+        let Self {
+            modulus_switched_glwe_ciphertext_list,
+            ciphertext_modulus,
+            message_modulus,
+            carry_modulus,
+            pbs_order,
+            lwe_per_glwe,
+            count: _,
```
[the remark this answers is not in the document]
Comment (mayeul-zama) [reply tfhe/src/shortint/ciphertext/compressed_ciphertext_list.rs]: Now checked

File: tfhe/src/core_crypto/entities/glwe_ciphertext.rs [hunk 3]
```diff
@@ -643,9 +643,19 @@ where
...
+
         check_encrypted_content_respects_mod(self, glwe_ct_parameters.ct_modulus)
-            && self.glwe_size() == glwe_ct_parameters.glwe_dim.to_glwe_size()
-            && self.polynomial_size() == glwe_ct_parameters.polynomial_size
-            && self.ciphertext_modulus() == glwe_ct_parameters.ct_modulus
+            && data.container_len()
+                == glwe_ct_parameters
+                    .glwe_dim
+                    .to_glwe_size()
+                    .to_glwe_ct_size(glwe_ct_parameters.polynomial_size)
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/entities/glwe_ciphertext.rs:657]: glwe_ciphertext_size

**Review by IceTDrinker — COMMENTED** [review]
small remark about the size primitive

**Review by IceTDrinker — APPROVED** [review]
Thanks ! This looks good and will be more robust to structure changes !

**Comment by IceTDrinker** [comment]
looks like some auto imports do not work

**Review by IceTDrinker — APPROVED** [review]

**Pull request merged**